### PR TITLE
Fix compiler warnings

### PR DESCRIPTION
**Pull request by JohannesLorenz**
Almost all warnings for gcc 10.2.0 and clang 11.0.0 are gone after this PR (together with #100 and #103).

### CODE REVIEW

**Comment by JohannesLorenz** [comment]
Compiler error detected in DPF (submodule updated), submitted as DISTRHO/DPF#247 .

File: src/UI/FormantFilterGraph.cpp [hunk 1]
```diff
@@ -63,7 +63,7 @@ void FormantFilterGraph::OSC_value(int x, const char *loc)
 }
 void FormantFilterGraph::OSC_value(unsigned x, void *v)
 {
-    assert(x = sizeof(Pvowels));
+    assert(x == sizeof(Pvowels));
```
Comment (fundamental) [inline src/UI/FormantFilterGraph.cpp:66]: Oh dear. That was a silly mistake of (likely) mine.

File: src/Misc/MiddleWare.cpp [hunk 2]
```diff
@@ -1096,7 +1096,7 @@ static int extractInt(const char *msg)
 
 static const char *chomp(const char *msg)
 {
-    while(*msg && *msg!='/') ++msg; \
+    while(*msg && *msg!='/') ++msg;
```
Comment (fundamental) [inline src/Misc/MiddleWare.cpp:1099]: Ah, that must have come from the old SNIP macro.

File: src/Synth/OscilGen.cpp [hunk 3]
```diff
@@ -269,7 +269,8 @@ const rtosc::MergePorts OscilGen::ports{
 //operations on FFTfreqs
 inline void clearAll(fft_t *freqs, int oscilsize)
 {
-    memset(freqs, 0, oscilsize / 2 * sizeof(fft_t));
+    fft_t zero = 0;
+    std::fill_n(freqs, oscilsize / 2, zero);
```
Comment (fundamental) [inline src/Synth/OscilGen.cpp:273]: I'd hope this turns into a memset under the hood. C++ standard should guarantee that std::complex<float> is the same as "_complex float" in C and that representation should be identical in memory as float[2]. Either way, it's better to fix the warning than fret about a few cycles difference.

**Comment by fundamental** [comment]
Seems good. Feel free to merge unless there's a desire to wait for the aforementioned DPF fix. (since tests are passing I'm guessing you selected some intermediary commit in that submodule.

File: src/Synth/OscilGen.cpp (hunk 3 above)
Comment (JohannesLorenz) [reply src/Synth/OscilGen.cpp:273]: I just found an interesting answer about exactly that on [StackOverflow](https://stackoverflow.com/a/51658588). It says we can either use `static_assert(std::is_trivially_copyable<std::complex<float>>::value);` before `memset`, or use `std::fill`, which is opitimized on GCC and clang, but not on MSVC. What do you think after reading this? I'm OK with either solution.

**Comment by JohannesLorenz** [comment]
Strange, it seems like just my CMake cache was outdated. I removed the CMake dir and rebuilt, and now it all works. Let me know about the `std::fill`, then I'll merge.

File: src/Synth/OscilGen.cpp (hunk 3 above)
Comment (fundamental) [reply src/Synth/OscilGen.cpp:273]: Well, as long as GCC/Clang are happy. AFAIK MSVC still hates the codebase due to parts of the rtosc port structure, so it's not really a supported dev platform anyhow.

**Pull request merged**